src/tests/frontend/specs/focus.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+describe('caret focus', function () {
+  describe('doesnt focus on editor using focusOnEditor URL Param', function () {
+    before(async function () {
+      await helper.aNewPad({
+        params: {
+          focusOnEditor: 'false'
+        }
+      });
+    });
+
+    it('does not focus when focusOnEditor is set to false', async function () {
+      const activeElement = helper.padChrome$.document.activeElement;
+      await helper.waitForPromise(() => $(activeElement).hasClass('readwrite'));
+    });
+
+  });
+
+  describe('does focus on editor using focusOnEditor URL Param', function () {
+    before(async function () {
+      await helper.aNewPad({
+        params: {
+          focusOnEditor: 'true'
+        }
+      });
+    });
+
+    it('does focus when focusOnEditor is set to true', async function () {
+      const activeElement = helper.padChrome$.document.activeElement;
+      console.log(activeElement)
+      await helper.waitForPromise(() => $(activeElement).attr('name') === 'ace_outer');
```

Comment:
It should already be focused by the time `helper.aNewPad()` resolves, so there's no need for `waitForPromise()` here. Also, shouldn't each frame's `activeElement` be inspected?

```suggestion
      expect(helper.padChrome$.document.activeElement)
          .to.be(helper.padChrome$.document.querySelector('iframe[name="ace_outer"]'));
      expect(helper.padOuter$.document.activeElement)
          .to.be(helper.padOuter$.document.querySelector('iframe[name="ace_inner"]'));
      expect(helper.padInner$.document.activeElement)
          .to.be(helper.padInner$.document.body);
```